test/test_mkldnn.py

```diff
@@ -514,6 +514,28 @@ def test_mul(self):
         torch.mul(mx, value, out=mkldnn_out)
         self.assertEqual(out, mkldnn_out.to_dense())
 
+    def test_0_dimension_tensor(self):
+        x = torch.rand([20, 20, 1, 1], dtype=torch.float)
+        y = torch.rand([20, 20, 0, 1], dtype=torch.float)
+
+        # unary ops work without modification
+        out_relu = torch.relu(y)
+        out_relu_mkldnn = torch.relu(y.to_mkldnn()).to_dense()
+        self.assertEqual(out_relu, out_relu_mkldnn)
+
+        out_mul = x * y
+        out_mul_mkldnn = (x.to_mkldnn() * y.to_mkldnn()).to_dense()
+        self.assertEqual(out_mul, out_mul_mkldnn)
+
+        out_add = x + y
+        out_add_mkldnn = (x.to_mkldnn() + y.to_mkldnn()).to_dense()
```

Comment:
might not be in the scope of this PR but what happens in case of conv and linear?

Reply:
this is in follow up PRs.